src/isl/util.rs

```diff
@@ -188,6 +191,39 @@ impl Range {
                 // TODO: Implement this section once the timestamp comparator for ion-rust is implemented
                 todo!()
             }
+            Range::TimestampPrecision(start, end) => {
+                let value = &value.as_timestamp().ok_or_else(|| {
+                    invalid_schema_error_raw(
+                        "Timestamp Precision ranges can only have timestamp value for validation",
+                    )
+                })?;
+
+                let value = TimestampPrecisionValue::from_timestamp(value);
+                let is_in_lower_bound = match start {
+                    Min => true,
+                    Value(start_value, boundary_type) => match start_value {
+                        TimestampPrecision(min_value) => match boundary_type {
+                            RangeBoundaryType::Inclusive => min_value <= &value,
+                            RangeBoundaryType::Exclusive => min_value < &value,
+                        },
+                        _ => unreachable!("TimestampPrecision range can only have timestamp precisions as lower and upper range boundary value"),
+                    },
+                    Max => unreachable!("Cannot have 'Max' as the lower range boundary")
+                };
+
+                let is_in_upper_bound = match end {
+                    Max => true,
+                    Min => unreachable!("Cannot have 'Min' as the upper range boundary"),
+                    Value(end_value, boundary_type) => match end_value {
+                        TimestampPrecision(max_value) => match boundary_type {
+                            RangeBoundaryType::Inclusive => max_value >= &value,
+                            RangeBoundaryType::Exclusive => max_value > &value,
+                        },
+                        _ => unreachable!("TimestampPrecision range can only have timestamp precisions as lower and upper range boundary value"),
+                    }
+                };
+                Ok(is_in_upper_bound && is_in_lower_bound)
+            }
```

Comment:
I notice that this method is getting to be ~200 lines long. It might be nice to break some of the `match` arms' logic out into helper methods.

Reply:
I will handle this when changing the Range implementation with #89.